react/week3/my-app/src/App.js

```diff
@@ -0,0 +1,148 @@
+import React from "react"
+import TodoItem from "./components/TodoItem"
+import todosData from "./components/todosData"
+import Timer from "./components/Timer"
+
+
+class App extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            todos: todosData,
+            //setTodos: todosData,
+            newItem: "",
+            deadline: ""
+        }
+
+        // const [todos, setTodos] = useState(todosDatas);
+        // this.setTodos = setTodos;
+        // this.todos = todos;
+
+        this.handleChange = this.handleChange.bind(this)
+        this.addTodo = this.addTodo.bind(this)
+        this.updateInput = this.updateInput.bind(this)
+        this.deleteItem = this.deleteItem.bind(this)
+    }
+
+    handleChange(id) {
+      this.setState((prevState) => {
+        const updatedTodos = prevState.todos.map((todo) => {
+          if (todo.id === id) {
+            return { ...todo, completed: !todo.completed };
+          } else {
+            return todo;
+          }
+        });
+        return { todos: updatedTodos };
+      });
+    }
+
+    addTodo(e) {
+      e.preventDefault();
+      const newTodo = {
+        id: this.state.todos.length + 1,
+        text: this.state.newItem,  // text should match with the input box
+        completed: false,
+        date: this.state.deadline
+      }
+      //const newTodo = this.state.todos;
+      const newTodos = this.state.todos.concat([newTodo]);
+      this.setState({
+        todos: newTodos
+      })
+    }
+
+    updateInput(value, id) { //original key, value
+    //  const items = this.state.todos;
+      // items.map(item => {
+      //   if(item.id === id) {
+      //     item.text=text;
+      //   }
+      // })
+
+      this.setState((prevState) => {
```

Comment:
Im not sure what happens here. This should only update the value of the create new todo input. Start looking here i would say